src/control/zoom/ZoomControl.cpp

```diff
@@ -18,7 +18,9 @@ auto onScrolledwindowMainScrollEvent(GtkWidget* widget, GdkEventScroll* event, Z
                 (event->direction == GDK_SCROLL_UP || (event->direction == GDK_SCROLL_SMOOTH && event->delta_y < 0)) ?
                         ZOOM_IN :
                         ZOOM_OUT;
-        zoom->zoomScroll(direction, {event->x, event->y});
+        int rx, ry;
+        gdk_window_get_root_coords(gtk_widget_get_window(widget), 0, 0, &rx, &ry);
+        zoom->zoomScroll(direction, {event->x_root - rx, event->y_root - ry});
```

Comment:
Please comment this piece! Some developers (like myself) have little experience with GTK! The documentation for `gdk_window_get_root_coords`. Why aren't we using `event->x` and `event->y`?

Reply:
> I replaced `event->x` with `event->x_root` as well, so instead of calculating the offset as double it calculates the center as integers independent of the scroll value. I don't understand where `event->x` comes from and replacing it in this way seemed to fix the page jump problem when zooming with ctrl+mousewheel. But on the other hand, from googling it seemed like `event->x_root` would be in window space, but it was in screen space for me...

Okay! It might be best to factor things related to this into a single place, so it's easier to change/read. Thank you!

Reply:
I don't know why, but reverting this seems to stop the view from jumping at the beginning of a control+wheel... It does, however, make the view scroll down while doing a ctrl+wheel...

Reply:
I have no experience with GTK as well. Where (which file) would you put the function for translating screen coordinates to window coordinates?

Reply:
I think [MainWindow](https://github.com/xournalpp/xournalpp/blob/master/src/gui/MainWindow.h) might be reasonable. I don't see anything like this already in that file, though...
It might also make sense to put this in something in `src/util`, but I don't see a file that provides similar/related functionality.

Reply:
Can we leave it as is for this PR and make a separate cleanup PR with no functional changes after this one merged? As I pointed out, this is not the only thing that needs cleanup.

Reply:
I think that's fine!